### PR TITLE
Add missing configuration and options source generator diagnostics

### PR DESCRIPTION
**Pull request by tarekgh**
---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/syslib-diagnostics/source-generator-overview.md](https://github.com/dotnet/docs/blob/4a1f938d63d2fb469b0dbe399a621d7a6bfedd29/docs/fundamentals/syslib-diagnostics/source-generator-overview.md) | [Source-generator diagnostics in .NET 6+](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/source-generator-overview?branch=pr-en-us-37686) |
| [docs/fundamentals/syslib-diagnostics/syslib1100-1118.md](https://github.com/dotnet/docs/blob/4a1f938d63d2fb469b0dbe399a621d7a6bfedd29/docs/fundamentals/syslib-diagnostics/syslib1100-1118.md) | [SYSLIB diagnostics for configuration binder source generation](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1100-1118?branch=pr-en-us-37686) |
| [docs/fundamentals/syslib-diagnostics/syslib1201-1219.md](https://github.com/dotnet/docs/blob/4a1f938d63d2fb469b0dbe399a621d7a6bfedd29/docs/fundamentals/syslib-diagnostics/syslib1201-1219.md) | [SYSLIB diagnostics for Microsoft.Extensions.Options.SourceGeneration](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1201-1219?branch=pr-en-us-37686) |

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @gewarren

File: docs/fundamentals/syslib-diagnostics/source-generator-overview.md [hunk 1]
```diff
@@ -138,6 +138,25 @@ The following table provides an index to the `SYSLIB1XXX` diagnostics in .NET 6
...
+| [SYSLIB1205][1205] | A type already includes an implementation of the 'Validate' method. |
+| [SYSLIB1206][1206] | Can't validate private fields or properties. |
+| [SYSLIB1207][1207] | Member type is not enumerable. |
+| [SYSLIB1208][1208] | Validators used for transitive or enumerable validation must have a constructor with no parameters. |
+| [SYSLIB1209][1209] | `OptionsValidatorAttribute` can't be applied to a static class. |
+| [SYSLIB1210][1210] | Null validator type specified for the `ValidateObjectMembersAttribute` or `ValidateEnumeratedItemsAttribute` attributes. |
+| [SYSLIB1211][1211] | Unsupported circular references in model types. |
+| [SYSLIB1212][1212] | Member potentially missing transitive validation. |
+| [SYSLIB1213][1213] | Member potentially missing enumerable validation. |
+| [SYSLIB1214][1214] | Can't validate constants, static fields or properties. |
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/source-generator-overview.md:154]:
```suggestion
| [SYSLIB1214][1214] | Can't validate constants, static fields, or properties. |
```

File: docs/fundamentals/syslib-diagnostics/syslib1100-1118.md [hunk 2]
```diff
@@ -12,3 +12,5 @@ The following table shows the diagnostic IDs for configuration binder source-gen
 |  `SYSLIB1100` | Type is not supported. |
 |  `SYSLIB1101` | Property on type is not supported. |
 |  `SYSLIB1102` | Project's language version must be at least C# 11. |
+|  `SYSLIB1103` | value types are invalid inputs to configuration 'Bind' methods. |
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib1100-1118.md:15]:
```suggestion
|  `SYSLIB1103` | Value types are invalid inputs to configuration 'Bind' methods. |
```

**Review by gewarren — APPROVED** [review]
Wow, thanks so much @tarekgh

**Pull request merged**